### PR TITLE
Serialize body more thoroughly, won't always be passed as kwarg

### PR DESCRIPTION
**Pull request by fabianvf**

### CODE REVIEW

File: openshift/dynamic/client.py [hunk 1]
```diff
@@ -163,6 +167,7 @@ def delete(self, resource, name=None, namespace=None, label_selector=None, field
 
     @meta_request
     def replace(self, resource, body=None, name=None, namespace=None, **kwargs):
+        body = self.serialize_body(body)
         body = body or {}
```
Comment (djzager) [inline openshift/dynamic/client.py]: Could this not just be `body = self.serialize_body(body) or {}` ?

File: openshift/dynamic/client.py [hunk 2]
```diff
@@ -174,7 +179,7 @@ def replace(self, resource, body=None, name=None, namespace=None, **kwargs):
 
     @meta_request
     def patch(self, resource, body=None, name=None, namespace=None, **kwargs):
-        body = body or {}
+        body = self.serialize_body(body)
```
Comment (djzager) [inline openshift/dynamic/client.py]: I'm not sure I understand the differences between how body is set for replace vs patch. What's up with that?

File: openshift/dynamic/client.py (hunk 1 above)
Comment (fabianvf) [reply openshift/dynamic/client.py]: actually line 171 should have been removed

Comment (fabianvf) [reply openshift/dynamic/client.py]: fixed

File: openshift/dynamic/client.py (hunk 2 above)
Comment (djzager) [reply openshift/dynamic/client.py]: Cool

**Pull request merged**